### PR TITLE
Version card

### PR DESCRIPTION
**Pull request by dfguerrerom**
related to #885 

- This PR creates a version card element that will be appended to the bottom of the `NavigationDrawer`.
- It will assume there's a `__version__.py` file in the root folder containing the current version
- It will assume the latest version of the module is in `release` branch
- If there's a release, it will use the text generated there as initial element, if not, only the changelog will be displayed.

[screen-capture.webm](https://github.com/12rambau/sepal_ui/assets/12363250/f17c1864-a7cf-414c-a397-2999d0e2f63a)

## WIP:
- I need to create the tests...
- But in the meantime, what do you think about this?

### CODE REVIEW

**Comment by 12rambau** [comment]
I like the idea of adding a version number but I would like not to use a __version__.py file. We already support the pyproject.toml file to upport [entry-point] key. I would suggest to use [[project.version]] for the version number. Advantage, it is supported by commitizen so it will be possible to update version with a simple cz bump.

File: sepal_ui/scripts/utils.py [hunk 1]
```diff
@@ -399,15 +400,12 @@ def get_app_version(github_url: str, branch="release") -> str:
     """
     repo_owner, repo_name = parse_github_url(github_url)
 
-    version_url = f"https://raw.githubusercontent.com/{repo_owner}/{repo_name}/{branch}/__version__.py"
-
+    version_url = f"https://raw.githubusercontent.com/{repo_owner}/{repo_name}/{branch}/pyproject.toml"
```
Comment (12rambau) [inline sepal_ui/scripts/utils.py:403]: I would not get the version number from the repository but from the application folder itself. Normally the folder is pulled from a repository and as such embed the .toml file. It ensures that the version displayed is consistent with what is installed on your machine and not what is available in distant repo. 
Also it avoids getting too much bound to github and start passively supporting other VCS like gitlab or azure devOps

**Comment by dfguerrerom** [comment]
@12rambau two things here:
- I've fixed the coverage issues we had, and now I think we'll have an increase.
- I don't really know why the builds are failing with the `Translator`, I didn't touch it.
- the readthedocs documentation failed because of a missing `toml` package... I'm not sure how readthedocs creates the documentation, are we giving the `[doc]` requirements to that env?

**Comment by 12rambau** [comment]
> I'm not sure how readthedocs creates the documentation, are we giving the [doc] requirements to that env?

yes we are in this file:  https://github.com/12rambau/sepal_ui/blob/main/.readthedocs.yaml
adding the toml deps in [doc] extra dependencies should do the trick.

File: sepal_ui/scripts/utils.py [hunk 2]
```diff
@@ -375,6 +374,94 @@ def check_input(input_: Any, msg: str = ms.utils.check_input.error) -> bool:
...
+        the version of the repository
+    """
+    # get the path to the pyproject.toml file
+    pyproject_path = repo_folder / "pyproject.toml"
+
+    # check if the file exist
+    if pyproject_path.exists():
+        # read the file using toml
+        with pyproject_path.open("r") as f:
+            pyproject = toml.load(f)
```
Comment (12rambau) [inline sepal_ui/scripts/utils.py:390]:
```suggestion
       pyproject = toml.loads(pyproject_path.read_text())
```

File: sepal_ui/sepalwidgets/app.py [hunk 3]
```diff
@@ -443,6 +437,7 @@ def __init__(
         code: str = "",
         wiki: str = "",
         issue: str = "",
+        repo_folder: str = Path.cwd(),
```
Comment (12rambau) [inline sepal_ui/sepalwidgets/app.py:440]: your hint expects a str and you serve a Pathlib.Path object. 

looking at the error in the code it seems it's also the source of the bug. Instead I would create somwhere (utils ?) a new Type: 
```python 
Pathlike = Union[str, pathlib.Path] 
```

and reuse it everytwhere we set file/dir names (to stay on the safe side). 

For the default value set it to `""` and fill it with Path.cwd() in the init method. At parsing stage, there is no cwd yet.

Comment (dfguerrerom) [reply sepal_ui/sepalwidgets/app.py]: Done, but I didn't use the `Pathlike` everywhere... perhaps is better if we handle that in a different PR? we could create a `types.py` file (if that's something common)

**Review by 12rambau — APPROVED** [review]
I'm happy with the latest change. As it's a rather old PR, I let you decide if you ant to add extra small things before merging.

**Pull request merged**